### ISSUE TITLE
Vertex AI: document URL sources rejected with incorrect error path

### ISSUE THREAD

**Issue by Avocher** [anthropics/anthropic-sdk-typescript#1038]
When using @anthropic-ai/vertex-sdk (v0.16.0) to send a document content block with a URL source, the Vertex AI endpoint rejects the request.

Payload:
```
  {
    "type": "document",
    "source": {
      "type": "url",
      "url": "https://..."
    },
    "cache_control": { "type": "ephemeral" },
    "citations": { "enabled": true }
  }
```

  Error returned:
  `messages.0.content.0.image.source.base64.data: URL sources are not supported`

  Issues:

1. URL sources are not supported on Vertex AI — the [Anthropic docs](https://platform.claude.com/docs/en/docs/build-with-claude/pdf-support) list URL sources as a supported option, but the Vertex endpoint rejects them.
2. The error path is wrong — the content block is a document with source type url, but the error references image.source.base64.data. This made the debugging difficult.

  Environment:
  - @anthropic-ai/vertex-sdk: 0.16.0
  - @anthropic-ai/sdk: 0.95.1

**Comment by dtmeadows** [comment]
Hey @Avocher I'm checking on this one internally. I was able to get this working if I pass the file directly as base64 though if you want to use that as a workaround for now: https://platform.claude.com/docs/en/build-with-claude/pdf-support#option-2-base64-encoded-pdf-document

**Comment by dtmeadows-ant** [comment]
@Avocher sorry for the slow follow-up here. Confirmed: URL sources (and Files API sources) for document and image blocks aren't available for Claude on Vertex AI today; only base64 sources are, and the platform docs now call that out: https://platform.claude.com/docs/en/build-with-claude/claude-on-vertex-ai#features-not-supported. On Vertex the approach is to fetch the file yourself and send it inline:

```ts
const res = await fetch('https://example.com/file.pdf');
const data = Buffer.from(await res.arrayBuffer()).toString('base64');

await client.messages.create({
  model: 'claude-sonnet-4-6',
  max_tokens: 1024,
  messages: [{
    role: 'user',
    content: [
      { type: 'document', source: { type: 'base64', media_type: 'application/pdf', data }, cache_control: { type: 'ephemeral' }, citations: { enabled: true } },
      { type: 'text', text: 'Summarize this document.' },
    ],
  }],
});
```
